### ISSUE TITLE
scss: no-unknown-animations: does not seem to trigger when duration is a variable

### ISSUE THREAD

**Issue by rburgst** [stylelint/stylelint#6512]
### What steps are needed to reproduce the bug?

I would expect the following scss file to produce an error, however, if you are using a variable for the timing then the rule does not seem to work: 

```scss
$duration: 1s;

.controls {
    animation: g-controls-fade-out $duration;
}
```

### What Stylelint configuration is needed to reproduce the bug?

```javascript
module.exports = {
    extends: ['stylelint-config-standard-scss', 'stylelint-config-prettier'],
    rules: {
        // See https://github.com/humanmade/coding-standards/pull/199/files
        'selector-class-pattern': [
            '^(?<block>(?:[a-z][a-z0-9]*)(?:-[a-z0-9]+)*)(?<element>(?:__[a-z][a-z0-9]*(?:-[a-z0-9]+)*))?(?<modifier>(?:_[a-z][a-z0-9]*)(?:-[a-z0-9]+)*)?$',
            {
                resolveNestedSelectors: true,
            },
        ],
        'no-unknown-animations': true,
    },
};

```

### How did you run Stylelint?

yarn stylelint **/*.scss

### Which version of Stylelint are you using?

`14.16.0`

### What did you expect to happen?

an error should be reported

### What actually happened?

no error is reported

### Does the bug relate to non-standard syntax?

seems to be related to SCSS variables

**Comment by jeddy3** [comment]
@rburgst Thanks for the report and for using the template.

The built-in rules are geared towards CSS. As such, most of them will ignore any constructs (_declaration_, _rules_ etc.) that contain non-standard syntax like dollar variables.

The rule [will flag the unknown animation name](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0kB2AXAThANgZwAJgAdVAggWgqgFdMBDdASwlXgIEY8BuU8+1EwC2jFmwIBzCmiy48FAGb1YFCDXQEAbvUwAKKrQbNWASl6oAviAA0IPAE8M9AB4JwePDfCsFTCW5IyAmIQTBocGDwQ9kDychDUCAoaVABrRIB3VAoBYVFWKIQCLBoYPgILUisLIA) if you use a standard CSS variable for the duration:

```css
.controls {
  --duration: 1s;
  animation: g-controls-fade-out var(--duration);
}
```

You could see if the [SCSS community who maintain configs and custom rules for SCSS](https://github.com/stylelint-scss) can create a version of the rule that supports SCSS syntax.
